osu.Game/OsuGame.cs

```diff
@@ -982,12 +982,15 @@ private void forwardLoggedErrorsToNotifications()
         /// <param name="component">The component to load.</param>
         /// <param name="loadCompleteAction">An action to invoke on load completion (generally to add the component to the hierarchy).</param>
         /// <param name="cache">Whether to cache the component as type <typeparamref name="T"/> into the game dependencies before any scheduling.</param>
-        private T loadComponentSingleFile<T>(T component, Action<T> loadCompleteAction, bool cache = false)
-            where T : Drawable
+        private T loadComponentSingleFile<T>(T component, Action<Drawable> loadCompleteAction, bool cache = false)
+            where T : class
         {
             if (cache)
                 dependencies.CacheAs(component);
 
+            var drawableComponent = component as Drawable;
+            Debug.Assert(drawableComponent != null);
```

Comment:
I'd probably make this a hard throw rather than an assertion. It should not be realistically possible for any of us to miss someone attempting to load a non-drawable via `loadComponentSingleFile()` (CI should catch it), but just in case, I wouldn't want these checks to elide on release builds and I would like the exception message to explain why this can't work.